src/AssetManager.php

```diff
@@ -0,0 +1,597 @@
+     *
+     * @return array the asset manager. Defaults to the "assetManager" application component.
+     */
+    public function getAssetBundles(): array
+    {
+        return $this->assetBundles;
+    }
+
+    public function getBasePath(): ?string
+    {
+        if (!empty($this->basePath)) {
+            $this->basePath = $this->aliases->get($this->basePath);
+        }
+
+        return $this->basePath;
+    }
+
+    public function getBaseUrl(): ?string
+    {
+        if (!empty($this->baseUrl)) {
+            $this->baseUrl = $this->aliases->get($this->baseUrl);
+        }
+
+        return $this->baseUrl;
+    }
+
+    /**
+     * Returns the named asset bundle.
+     *
+     * This method will first look for the bundle in {@see bundles()}. If not found, it will treat `$name` as the class
+     * of the asset bundle and create a new instance of it.
+     *
+     * @param string $name the class name of the asset bundle (without the leading backslash).
+     *
+     * @return AssetBundle the asset bundle instance
+     *
+     * @throws \InvalidArgumentException
+     * @throws InvalidConfigException
+     */
+    public function getBundle(string $name): AssetBundle
+    {
+        if (!isset($this->bundles[$name])) {
+            return $this->bundles[$name] = $this->publish->loadBundle($name, []);
+        }
+        if ($this->bundles[$name] instanceof AssetBundle) {
+            return $this->bundles[$name];
+        }
+        if (\is_array($this->bundles[$name])) {
+            return $this->bundles[$name] = $this->publish->loadBundle($name, $this->bundles[$name]);
+        }
+        if ($this->bundles[$name] === false) {
+            return $this->loadDummyBundle($name);
+        }
+        throw new \InvalidArgumentException("Invalid asset bundle configuration: $name");
+    }
+
+    /**
+     * Returns the asset converter.
+     *
+     * @return AssetConverterInterface the asset converter.
+     */
+    public function getConverter(): AssetConverterInterface
+    {
+        if (empty($this->converter)) {
+            $this->converter = new AssetConverter($this->aliases, $this->logger);
+        }
+
+        return $this->converter;
+    }
+
+    public function getCssFiles(): array
+    {
+        return $this->cssFiles;
+    }
+
+    public function getDirMode(): int
+    {
+        return $this->dirMode;
+    }
+
+    public function getFileMode(): int
+    {
+        return $this->fileMode;
+    }
+
+    public function getForceCopy(): bool
+    {
+        return $this->forceCopy;
+    }
+
+    public function getJsFiles(): array
+    {
+        return $this->jsFiles;
+    }
+
+    public function getLinkAssets(): bool
+    {
+        return $this->linkAssets;
+    }
+
+    public function getHashCallback(): ?callable
+    {
+        return $this->hashCallback;
+    }
+
+    public function getPublish(): AssetPublisher
+    {
+        if (empty($this->publish)) {
+            $this->publish = new AssetPublisher($this);
```

Comment:
Remember that this extension is used in the view, and the more dependencies you pass in the constructor, the more dependencies you must load in view, so only aliases and logger are used that are already loaded in the same view with getAliases() and getLogger().

Reply:
AssetManager as well as AssetPublisher and AssetConverter will not created directly using `new Asset*`. Right? If required it will be get via `$container->get(*)` or via DI from `__construct`.
So it can and should be configured via DI.